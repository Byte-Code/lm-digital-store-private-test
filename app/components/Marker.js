import React, { PropTypes } from 'react';
import PlaceIcon from 'material-ui/svg-icons/maps/place';
import MyPlaceIcon from 'material-ui/svg-icons/maps/person-pin-circle';
import glamorous from 'glamorous';

const Wrapper = glamorous.div({
  position: 'relative',
  zIndex: 1,
  transform: 'translate(-50%, -50%)'
});

const WhiteBg = glamorous.div({
  position: 'absolute',
  height: '25px',
  width: '20px',
  zIndex: -1,
  top: '9px',
  left: '16px',
  backgroundColor: '#fff'
});


const Marker = ({ handleClick, isCurrentStore, isAvailable }) => {
  const Icon = isCurrentStore ? MyPlaceIcon : PlaceIcon;
  const iconColor = isAvailable ? '#67cb33' : '#000';
  return (
    <Wrapper>
      <Icon
        style={{ height: 55, width: 55, cursor: 'pointer' }}
        color={iconColor}
        onTouchTap={handleClick}
      />
      <WhiteBg />
    </Wrapper>
  );
};

Marker.propTypes = {
  isCurrentStore: PropTypes.bool.isRequired,
  isAvailable: PropTypes.bool.isRequired,
  handleClick: PropTypes.func.isRequired
};

export default Marker;
