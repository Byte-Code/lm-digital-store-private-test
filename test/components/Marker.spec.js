import React from 'react';
import { shallow } from 'enzyme';
import PlaceIcon from 'material-ui/svg-icons/maps/place';
import MyPlaceIcon from 'material-ui/svg-icons/maps/person-pin-circle';

import Marker, { Wrapper } from '../../app/components/Marker';

const handleClick = jest.fn();

describe('Marker', () => {
  it('should render with a PlaceIcon if isCurrentStore is false', () => {
    const wrapper = shallow(
      <Marker
        handleClick={handleClick}
        isCurrentStore={false}
        isAvailable={false}
      />
    );
    expect(wrapper.find(PlaceIcon)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a MyPlaceIcon if isCurrentStore is true', () => {
    const wrapper = shallow(
      <Marker
        handleClick={handleClick}
        isCurrentStore
        isAvailable={false}
      />
    );
    expect(wrapper.find(MyPlaceIcon)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a black icon if isAvailable is false', () => {
    const wrapper = shallow(
      <Marker
        handleClick={handleClick}
        isCurrentStore={false}
        isAvailable={false}
      />
    );
    expect(wrapper.find(PlaceIcon).prop('color')).toEqual('#000');
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a green icon if isAvailable is false', () => {
    const wrapper = shallow(
      <Marker
        handleClick={handleClick}
        isCurrentStore={false}
        isAvailable
      />
    );
    expect(wrapper.find(PlaceIcon).prop('color')).toEqual('#67cb33');
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleClick when clicked', () => {
    const wrapper = shallow(
      <Marker
        handleClick={handleClick}
        isCurrentStore={false}
        isAvailable
      />
    );
    wrapper.find(Wrapper).simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });
});
