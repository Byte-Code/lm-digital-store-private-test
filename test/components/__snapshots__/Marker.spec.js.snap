// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker should render a black icon if isAvailable is false 1`] = `
<glamorous(div)
  onClick={[Function]}
>
  <MapsPlace
    color="#000"
    style={
      Object {
        "cursor": "pointer",
        "height": 55,
        "width": 55,
      }
    }
  />
  <glamorous(div) />
</glamorous(div)>
`;

exports[`Marker should render a green icon if isAvailable is false 1`] = `
<glamorous(div)
  onClick={[Function]}
>
  <MapsPlace
    color="#67cb33"
    style={
      Object {
        "cursor": "pointer",
        "height": 55,
        "width": 55,
      }
    }
  />
  <glamorous(div) />
</glamorous(div)>
`;

exports[`Marker should render with a MyPlaceIcon if isCurrentStore is true 1`] = `
<glamorous(div)
  onClick={[Function]}
>
  <MapsPersonPinCircle
    color="#000"
    style={
      Object {
        "cursor": "pointer",
        "height": 55,
        "width": 55,
      }
    }
  />
  <glamorous(div) />
</glamorous(div)>
`;

exports[`Marker should render with a PlaceIcon if isCurrentStore is false 1`] = `
<glamorous(div)
  onClick={[Function]}
>
  <MapsPlace
    color="#000"
    style={
      Object {
        "cursor": "pointer",
        "height": 55,
        "width": 55,
      }
    }
  />
  <glamorous(div) />
</glamorous(div)>
`;
